src/main/java/org/apache/commons/text/names/HumanNameParser.java

```diff
@@ -131,20 +135,24 @@ public Name parse(String name) {
 
         NameString nameString = new NameString(name);
         // TODO compile regexes only once when the parser is created
+        String salutations = StringUtils.join(this.salutations, " |") + "";
         String suffixes = StringUtils.join(this.suffixes, "\\.*|") + "\\.*";
         String prefixes = StringUtils.join(this.prefixes, " |") + " ";
 
         // The regex use is a bit tricky.  *Everything* matched by the regex will be replaced,
         // but you can select a particular parenthesized submatch to be returned.
         // Also, note that each regex requres that the preceding ones have been run, and matches chopped out.
         // names that starts or end w/ an apostrophe break this
+        String salutationRegex = "^(?i)(("+salutations+")\\.)";
```

Comment:
Our specific use case has no `.` but otherwise works well with this library. We'd much prefer that it supports both "Dr" and "Dr.".